doc/source/general/cpu_hardware.rst

```diff
@@ -142,40 +146,134 @@ Triggers are all of the following:
 -  ON condition { some commands }.
 -  WHEN condition THEN { some commands }.
 
-.. note::
-
-    The :ref:`WAIT <wait>` command only causes mainline code
-    to be suspended.  Trigger code such as WHEN, ON, LOCK STEERING,
-    and LOCK THROTTLE, will continue executing while your program
-    is sitting still on the WAIT command.
+The way these work is that once per **physics tick**, all these
+trigger routines get run, including those locks that are always
+re-evaluated by the cooked steering, and the ``ON`` and ``WHEN``
+triggers.  (This isn't *quite* true.  The real answer is more
+complex than that - see :ref:`CPU Update Loop <cpu_update_loop>`
+elsewhere on this page).
+
+Each of the steering locks behaves like a function that returns
+a value, and is re-called to get the new value for this **physics
+tick**.  Each of the ``ON`` and ``WHEN`` triggers also behave
+much like a function, with a body like this::
+
+   if (not conditional_expression)
+       return 1.  // premature quit.  preserve and try again next time.
```

Comment:
should this be `return true.` instead?  So as not to imply that `return 2.` might not mean "premature quit"?